Editor/PipeEditor.cs

```diff
@@ -64,10 +65,12 @@ public override void OnInspectorGUI()
 
             var hasCaps = EditorGUILayout.ToggleLeft("End Caps", _generator.HasCaps, EditorStyles.boldLabel);
             float capRadius = _generator.CapRadius, capThickness = _generator.CapThickness;
+            float capOffset = _generator.CapOffset;
             if (_generator.HasCaps)
             {
                 capRadius = EditorGUILayout.Slider("Radius", _generator.CapRadius, 0, radius);
                 capThickness = EditorGUILayout.Slider("Thickness", _generator.CapThickness, 0, radius);
+                capOffset = EditorGUILayout.Slider("Offset", _generator.CapOffset, 0, radius * 4);
```

Comment:
This brings up a more general "problem" I'm thinking about how to solve. If you want to create e.g. electrical conduit, the "rings" tend to be longer rather than thicker. But if they are too long (I was messing with the code and increased the limits) and connecting a curved section, they start to intersect the pipe. One solution would be to offset along the straight section. The offset for the end caps could run into a similar problem, but the `Height` should probably take care of that, and would also help with the other one.

I'll fix this one after experimenting a bit to make sure it works well and is easy to use.

Reply:
Yeah, the current solution is tailored for a very specific type of pipe. I though of adding stuff for other types (for example, curved joints like on plastic pipes), but there's a possible trade-off of overcomplicating both the code and the UX, so I just stuck to the most generally used one. Hopefully you can find a simple solution for what you're trying to do :)